src/file/imp/windows.rs

```diff
@@ -92,22 +92,22 @@ pub fn persist(old_path: &Path, new_path: &Path, overwrite: bool) -> io::Result<
     // TODO: We should probably do this in one-shot using SetFileInformationByHandle but the API is
     // really painful.
 
-    unsafe {
-        let old_path_w = to_utf16(old_path);
-        let new_path_w = to_utf16(new_path);
+    let old_path_w = to_utf16(old_path);
+    let new_path_w = to_utf16(new_path);
```

Comment:
As the types here do not ensure that these vectors are safe to pass into the kernel as c strings, this should go in the unsafe block.